### PR TITLE
Add grapple teleport to schema with example

### PR DESCRIPTION
**Pull request by blkerby**

### CODE REVIEW

File: tech.json [hunk 1]
```diff
@@ -465,6 +465,18 @@
...
               "devNote": "Bombs is a leniency as the trick has a high chance of failure."
+            },
+            {
+              "name": "canGrappleTeleport",
+              "requires": [
+                "canUseGrapple"
+              ],
+              "note": [
+                "Exiting a door transition while grappled, in order to initiate a teleport in the next room.",
+                "The position to which Samus teleports corresponds to the position of the grapple block in the first room.",
```
[the remark this answers is not in the document]
Comment (osse101) [reply tech.json:476]: I haven't seen if there is another way other than with Moonwalk.

File: schema/m3-room.schema.json [hunk 2]
```diff
@@ -486,6 +486,32 @@
...
+            },
+            "comeInWithGrappleTeleport": {
+              "$id": "#/definitions/strat/properties/entranceCondition/properties/comeInWithGrappleTeleport",
+              "type": "object",
+              "description": "Represents that Samus must come into the room while grappling, teleporting Samus to a position in this room corresponding to the location of the (grapple) block in the other room.",
+              "required": ["blockPositions"],
+              "additionalProperties": false,
+              "properties": {
+                "blockPositions": {
+                  "$id": "#/definitions/strat/properties/exitCondition/properties/leaveWithGrappleTeleport/properties/blockPositions",
```
Comment (osse101) [inline schema/m3-room.schema.json:498]:
```suggestion
                  "$id": "#/definitions/strat/properties/entranceCondition/properties/comeInWithGrappleTeleport/properties/blockPositions",
```

File: schema/m3-room.schema.json [hunk 3]
```diff
@@ -486,6 +486,32 @@
...
+                  "type": "array",
+                  "description": "A list of tile coordinates of (grapple) blocks that Samus could be grappled to while exiting the other room.",
+                  "items": {
+                    "$id": "#/definitions/strat/properties/exitCondition/properties/leaveWithGrappleTeleport/properties/blockPositions/items",
```
Comment (osse101) [inline schema/m3-room.schema.json:502]:
```suggestion
                    "$id": "#/definitions/strat/properties/entranceCondition/properties/comeInWithGrappleTeleport/properties/blockPositions/items",
```

File: schema/m3-room.schema.json [hunk 4]
```diff
@@ -486,6 +486,32 @@
...
+                    "type": "array",
+                    "description": "Coordinates of a (grapple) block that Samus could be grappled to while exiting the other room.",
+                    "minItems": 2,
+                    "maxItems": 2,
+                    "items": {
+                      "$id": "#/definitions/strat/properties/exitCondition/properties/leaveWithGrappleTeleport/properties/blockPositions/items/items",
```
Comment (osse101) [inline schema/m3-room.schema.json:508]:
```suggestion
                      "$id": "#/definitions/strat/properties/entranceCondition/properties/comeInWithGrappleTeleport/properties/blockPositions/items/items",
```

File: tech.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (blkerby) [reply tech.json:476]: Bobbob mentioned that most of the setups involve glitched grapple hanging: https://www.youtube.com/watch?v=Dm56Ev7No1Y

[the remark this answers is not in the document]
Comment (blkerby) [reply tech.json:476]: Based on this description, I added a note to the Red Brinstar Fireflea strat now to explain about moonwalking on the same frame that Grapple attaches

**Pull request merged**